### PR TITLE
followup from review comments for #2709

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

This PR addresses these two comments from #2709:

- [This comment about using the suffix `Fn`](https://github.com/cosmos/ibc-go/pull/2709#discussion_r1024070188).
- [This comment about adding `doc.go` for packages](https://github.com/cosmos/ibc-go/pull/2709#discussion_r1080734398).

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by crodriguezvega** [comment]
I didn't rename the functions in the test cases struct (i.e. `malleate`, `expResult`...) because there are a lot of them 😓 but I can do it if people still think is good for consistency.

File: modules/light-clients/07-tendermint/store.go [hunk 1]
```diff
@@ -280,7 +280,7 @@ func PruneAllExpiredConsensusStates(
 ) int {
 	var heights []exported.Height
 
-	pruneCb := func(height exported.Height) bool {
+	pruneCbFn := func(height exported.Height) bool {
```
Comment (damiannolan) [inline modules/light-clients/07-tendermint/store.go:283]: I don't think we should enforce using the `Fn` suffix. I don't think it provides too much value, and for a lot of cases its usually pretty clear from naming that its a function. E.g. here we're using `cb` for callback, where a callback is always a function. Similarly in the ante handler `next` is standard naming conventions for middleware patterns like the ante handler chain so I think `nextFn` doesn't do much. 

That's just my opinion anyways, I would prefer to not be super strict about this style choice! :)

File: modules/core/05-port/doc.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/*
+Package port implements the ICS 05 - Port Allocation specification
+https://github.com/cosmos/ibc/tree/main/spec/core/ics-005-port-allocation. This
```
Comment (damiannolan) [inline modules/core/05-port/doc.go:3]:
```suggestion
Package port implements the [ICS 05 - Port Allocation specification]
(https://github.com/cosmos/ibc/tree/main/spec/core/ics-005-port-allocation). This
```

File: modules/core/04-channel/doc.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/*
+Package channel implements the [ICS 04 - Channel and Packet Semantics specification]
+(https://github.com/cosmos/ibc/tree/main/spec/core/ics-004-channel-and-packet-semantics). This
+concrete implementation defines types and methods for safely creating two
+stateful objects (channels) on two separate chains, each associated with a
```
Comment (damiannolan) [inline modules/core/04-channel/doc.go:5]:
```suggestion
stateful objects (channel ends) on two separate chains, each associated with a
```

File: modules/apps/transfer/doc.go [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/*
+Package tansfer implements the packet data structure, state machine handling logic,
```
Comment (damiannolan) [inline modules/apps/transfer/doc.go:2]:
```suggestion
Package transfer implements the packet data structure, state machine handling logic,
```

File: modules/apps/29-fee/doc.go [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/*
+Package fee implements the packet data structure, state machine handling logic,
+and encoding details for handling fee payments on top of any ICS application protocol.
+This implementation is based off the [ICS 29 specification]
+(https://github.com/cosmos/ibc/tree/main/spec/app/ics-029-fee-payment)
```
Comment (damiannolan) [inline modules/apps/29-fee/doc.go:5]: Should we also say its a middleware implementation, ICS30?

File: modules/apps/27-interchain-accounts/doc.go [hunk 6]
```diff
@@ -0,0 +1,8 @@
+/*
+Package ica implements the packet data structure, state machine handling logic,
+and encoding details for the account management system over an IBC channel
+between separate chains.
+This implementation is based off the [ICS 29 specification]
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/doc.go:5]:
```suggestion
between separate chains.
This implementation is based off the [ICS 27 specification]
```

**Review by damiannolan — COMMENTED** [review]
Awesome, I just left some smalls nits/corrections. 
The doc.go files look pretty nice, are we missing one for 23-commitment?

File: modules/light-clients/07-tendermint/store.go (hunk 1 above)
Comment (crodriguezvega) [reply modules/light-clients/07-tendermint/store.go:283]: Sounds good to me, @damiannolan. I will change it.

**Review by chatton — APPROVED** [review]
LGTM the only thing that would be nice to double check is if the markdown syntax renders correctly when go docs are generated.

**Comment by crodriguezvega** [comment]
> LGTM the only thing that would be nice to double check is if the markdown syntax renders correctly when go docs are generated.

Looks like the MD is not rendered properly, so I will change that.

**Review by damiannolan — APPROVED** [review]

**Pull request merged**